libs/dh/metering-point/feature-overview/src/lib/dh-customer-overview.component.ts

```diff
@@ -67,46 +69,56 @@ import { DhCustomerContactDetailsComponent } from './dh-customer-contact-details
         <h3>{{ t('title') }}</h3>
       </watt-card-title>
 
-      <div
-        vater-stack
-        direction="row"
-        gap="s"
-        class="protected-address watt-space-inset-squish-s watt-space-stack-m"
-      >
-        <watt-icon size="s" name="warning" />
-        <span class="watt-text-s">{{ t('protectedAddress') }}</span>
-      </div>
-
       <div vater-flex gap="m" direction="row" class="watt-space-stack-m">
         @for (customer of customers(); track customer.id) {
           <div vater-flex gap="s" basis="0" class="customer">
+            <div
+              vater-stack
+              direction="row"
+              gap="s"
+              class="watt-space-inset-squish-s watt-space-stack-m"
+              [class.protected-address]="customer.isProtectedName"
+            >
+              @if (customer.isProtectedName) {
+                <watt-icon size="s" name="warning" />
+                <span class="watt-text-s">{{ t('protectedAddress') }}</span>
+              }
+            </div>
+
             <h5>{{ customer.name }}</h5>
             <dh-customer-cpr [customerId]="customer.id" />
           </div>
         }
       </div>
 
-      <a (click)="$event.preventDefault(); showAddressDetails()" class="watt-link-s">{{
-        t('showContactDetailsLink')
-      }}</a>
+      @if (showContactDetails()) {
+        <a (click)="$event.preventDefault(); openContactDetails()" class="watt-link-s">{{
+          t('showContactDetailsLink')
+        }}</a>
+      }
     </watt-card>
   `,
 })
 export class DhCustomerOverviewComponent {
-  modalService = inject(WattModalService);
+  private modalService = inject(WattModalService);
+
+  meteringPoint = input.required<MeteringPointDetails | undefined>();
+
+  customers = computed(() =>
+    this.meteringPoint()?.currentCommercialRelation?.currentEnergySupplierPeriod?.contacts.filter(
+      (x) =>
+        x.relationType === CustomerRelation.Primary || x.relationType === CustomerRelation.Secondary
+    )
```

Comment:
question: shouldn't there be a `[]` fallback in case of `undefined`? Strange that TS does not flag that.

Reply:
Good idea